source/GameData.cpp

```diff
@@ -187,7 +187,15 @@ void GameData::LoadShaders(bool useShaderSwizzle)
 
 double GameData::GetProgress()
 {
-	return min(min(spriteQueue.GetProgress(), Audio::GetProgress()), objects.GetProgress());
+	// Since the game defers loading of certain sprites we need to cache
+	// the progress so that when it reaches 1. (when the initial data has
+	// been loaded) it won't ever go back down.
+	static double progress;
```

Comment:
```suggestion
	// Cache the maximum progress seen, so clients are
	// isolated from the loading implementation details.
	static double progress = 0.;
```
We should ensure the double is always initialized correctly.

Reply:
The variable has static storage duration and as such it will be initialized with 0 just like globals). Anyways, I'll think I'll use the approach mentioned by peter (the one used originally). I considered it when I was writing this PR but didn't like it for some reason I can't remember.